### PR TITLE
Make executing 'undefined builds' an error

### PR DESCRIPTION
**Pull request by jjohannes**
Also delete the project cache dir in the folder that turned out not to be a Gradle project.

See also #13793

### CODE REVIEW

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

File: subprojects/core/src/main/java/org/gradle/initialization/layout/ProjectCacheDir.java [hunk 1]
```diff
@@ -43,8 +48,20 @@ public File getDir() {
...
     @Override
     public void stop() {
+        if (deleteOnStop) {
+            try {
+                deleter.deleteRecursively(dir);
+            } catch (IOException e) {
+                LOGGER.debug("Failed to delete unused project cache dir " + dir.getAbsolutePath(), e);
+            }
+            return;
+        }
```
Comment (donat) [inline subprojects/core/src/main/java/org/gradle/initialization/layout/ProjectCacheDir.java:64]: Shouldn't we handle this as part of `VersionSpecificCacheCleanupAction`? If the user accidentally deletes the settings file and runs a task, will he lose all the caches?

**Review by donat — APPROVED** [review]
Looks good overall. Should we add something to the getting started guide?

File: subprojects/core/src/main/java/org/gradle/initialization/layout/ProjectCacheDir.java (hunk 1 above)
Comment (jjohannes) [reply subprojects/core/src/main/java/org/gradle/initialization/layout/ProjectCacheDir.java:64]: The idea here is that in the end there will be no ".gradle" folder left somewhere on your disc where you accidentally started Gradle. 

Another option would to just not care. But I also think it is not such a big deal to "loose" the history of the project cache.   The most valuable caches are in the user home anyway. And I am not sure that a scenario where the user deletes the file by mistake and then runs Gradle is very common.

**Comment by jjohannes** [comment]
> Should we add something to the getting started guide?

I don't think this is necessary. It's an error case you should not run into usually. And if you do, the idea is that the error message gives you enough information. Or did you think of something specific we currently miss?

File: subprojects/core/src/main/java/org/gradle/initialization/layout/ProjectCacheDir.java (hunk 1 above)
Comment (donat) [reply subprojects/core/src/main/java/org/gradle/initialization/layout/ProjectCacheDir.java:64]: 👍

**Comment by donat** [comment]
No, I think it's all good. Just wanted to mention it.

**Pull request merged**